app/src/auto_validator/core/tasks.py

```diff
@@ -1,12 +1,62 @@
+import bittensor as bt
 import structlog
 from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.core.cache import cache
 
 from auto_validator.celery import app
 
+from .models import SubnetSlot, ValidatorInstance
+
 logger = structlog.wrap_logger(get_task_logger(__name__))
 
 
 @app.task
 def demo_task(x, y):
     logger.info("adding two numbers", x=x, y=y)
     return x + y
+
+
+@app.task
+def update_validator_status():
+    # Step 1: Get all subnet slots ordered by ID
+    subnet_slots = SubnetSlot.objects.filter().order_by("id")
+    total_slots = subnet_slots.count()
+
+    if total_slots == 0:
+        logger.warning("No subnet slots found.")
+        return
+
+    # Step 2: Get the current slot index from the cache (default to 0)
+    current_index = cache.get("current_subnet_slot_index", 0)
+
+    # Step 3: Get the next subnet slot to process
+    slot = subnet_slots[current_index]
+    blockchain = slot.blockchain
+    netuid = slot.netuid
+    # Step 4: Process the selected slot
+    try:
+        subtensor = bt.subtensor(network=settings.BT_NETWORK_NAME)
+        validators = ValidatorInstance.objects.filter(subnet_slot=slot)
+        if validators.exists():
+            metagraph = bt.metagraph(netuid=netuid, network="local", lite=True, sync=True)
```

Comment:
network="local" won't work - here we need to use a subtensor which has an address configured in settings, but also for testnet that won't work either, so we need to use the testnet subtensor provided by OTF

Reply:
also this operation fails a lot so implement basic retries